File: languages/js/src/entity/integrity/anchor_network.ts
import * as proto from "../../bridge/proto/bloock_integrity_entities";

/**
 * Represents information about an anchor network.
 */
export class AnchorNetwork {
  name: string;
  state: string;
  txHash: string;
  root?: string;

  /**
   * Constructs an AnchorNetwork object with the specified parameters.
   * @param name
   * @param state
   * @param txHash
   * @param root
   */
  constructor(name: string, state: string, txHash: string, root?: string) {
    this.name = name;
    this.state = state;
    this.txHash = txHash;
    this.root = root;
  }

  static fromProto(a: proto.AnchorNetwork): AnchorNetwork {
    return new AnchorNetwork(a.name, a.state, a.txHash, a.root);
  }

  toProto(): proto.AnchorNetwork {
    return proto.AnchorNetwork.fromPartial({
      name: this.name,
      state: this.state,
      txHash: this.txHash,
      root: this.root
    });
  }
}

File: languages/go/entity/integrity/anchor_network.go
package integrity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// AnchorNetwork represents information about an anchor network.
type AnchorNetwork struct {
	Name   string
	State  string
	TxHash string
	Root   *string
}

func NewAnchorNetworkFromProto(a *proto.AnchorNetwork) AnchorNetwork {
	if a == nil {
		return AnchorNetwork{}
	}
	return AnchorNetwork{
		Name:   a.Name,
		State:  a.State,
		TxHash: a.TxHash,
		Root:   a.Root,
	}
}

func MapAnchorNetworksFromProto(n []*proto.AnchorNetwork) []AnchorNetwork {
	networks := make([]AnchorNetwork, len(n))
	for i, network := range n {
		networks[i] = NewAnchorNetworkFromProto(network)
	}
	return networks
}

func (a AnchorNetwork) ToProto() *proto.AnchorNetwork {
	return &proto.AnchorNetwork{
		Name:   a.Name,
		State:  a.State,
		TxHash: a.TxHash,
		Root:   a.Root,
	}
}
